### PR TITLE
NEW: Can invoice task time per different services

### PR DESCRIPTION
**Pull request by FHenry**
Issue : #19763

Follow #19764

### CODE REVIEW

**Comment by isamuse2** [comment]
Very helpful feature widely anticipated

File: htdocs/projet/tasks/time.php [hunk 1]
```diff
@@ -268,18 +271,20 @@
...
+				}
 			}
 
-			if ($result >= 0) {
-				setEventMessages($langs->trans("RecordSaved"), null, 'mesgs');
-			} else {
-				setEventMessages($langs->trans($object->error), null, 'errors');
-				$error++;
-			}
+			$result = $object->addTimeSpent($user);
```
Comment (eldy) [inline htdocs/projet/tasks/time.php:287]: It seems this line is executed twice. 
It is already executed before (with a condition on authorization to do it). I think we must remove this line (that is done without authorization control). Can you confirm ?

**Comment by lmag** [comment]
:)

**Comment by eldy** [comment]
@FHenry The label of PR still contains the "WIP" in title. True or not ? If no more a WIP, can you merge it with last develop ?

**Comment by FHenry** [comment]
@eldy Yes still so not merge Work in progress, it's almost done but I have to loop on each  line for get product data whatever invoice mode selected. 
I made a mistake on first implementation where I add a new invoice mode, but that's silly.
If product is set on task_time I have to invoice cartesian product of  product*user or product/task(heuu not this one), product*time

**Comment by lmag** [comment]
we can work on it next friday ?

**Comment by FHenry** [comment]
@eldy for me its ok now

**Comment by lmag** [comment]
Un fail pour @FHenry qui n'est pas de pour lui :)

**Pull request merged**